server/routes/users.js

```diff
@@ -49,19 +49,25 @@ export default (storage, scriptManager) => {
         user: req.user
       },
       payload: {
-        search: req.query.search
+        search: req.query.search,
+        filterBy: req.query.filterBy
       }
     };
 
+    let searchQuery = req.query.search;
+    if (req.query.filterBy && req.query.filterBy.length > 0) {
+      searchQuery = `${req.query.filterBy}:"${req.query.search}"`;
+    }
+
     scriptManager.execute('filter', filterContext)
       .then((filter) => {
         const options = {
           sort: 'last_login:-1',
-          q: (req.query.search && filter) ? `(${req.query.search}) AND ${filter}` : req.query.search || filter,
+          q: (searchQuery && filter) ? `(${searchQuery}) AND ${filter}` : searchQuery || filter,
           per_page: req.query.per_page || 10,
           page: req.query.page || 0,
           include_totals: true,
-          fields: 'user_id,name,email,identities,picture,last_login,logins_count,multifactor,blocked,app_metadata',
+          fields: 'user_id,username,name,email,identities,picture,last_login,logins_count,multifactor,blocked,app_metadata',
```

Comment:
interesting.  I think we need to be building the list of fields from the list of userFields.  Some static default set, and then other fields based on the custom fields they are creating.  For example user_metadata isn't here, it probably should be, or should be dynamically added based on the custom fields set.